src/Streamly/Coreutils/FileTest.hs

```diff
@@ -306,6 +335,29 @@ isSocket = predicate Files.isSocket
 isTerminalFD :: FileTest
 isTerminalFD = undefined
 
+#endif
+
+-- | True if the file is writable for the current user.
+--
+-- Note that the file is not writable on a read-only file system even if this
+-- test indicates true.
+--
+-- Like coreutil @test -w file@
+--
+-- /Pre-release/
+isWritable :: FileTest
+isWritable =
+#if !defined(CABAL_OS_WINDOWS)
+    hasPermissions
+        (
+          Files.ownerWriteMode
+        , Files.groupWriteMode
+        , Files.otherWriteMode
+        )
+#else
+    predicateM $ \_ -> return True
```

Comment:
Always writable on Windows?

Reply:
I could not find a way to convert 
fileAccess :: FilePath -> Bool -> Bool -> Bool -> IO Bool  as FileTest